config/config.py

```diff
@@ -71,32 +70,65 @@ def set_debug_mode(self, value: bool) -> None:
         """Set the debug mode value."""
         self.debug_mode = value
 
+class APIKeyError(Exception):
+    """
+    Exception raised when an API key is not set in config.py or as an environment variable.
+    """
+    def __init__(self, service_name: str):
+        self.service_name = service_name
+
+    def __str__(self):
+        if self.service_name == "Tavily":
+            service_env = "TAVILY_API_KEY"
+            link = "https://app.tavily.com"
+        elif self.service_name == "GoogleSerp":
+            service_env = "SERP_API_KEY"
+            link = "https://serper.dev/"
+        elif self.service_name == "Google":
+            service_env = "Google_API_KEY and GOOGLE_CX"
+            link = "https://developers.google.com/custom-search/v1/overview"
+        elif self.service_name == "OpenAI":
+            link = "https://platform.openai.com/account/api-keys"
+            return (
+                Fore.RED
+                + "Please set your OpenAI API key in .env or as an environment variable.\n"
+                + "You can get your key from https://platform.openai.com/account/api-keys"
+            )
+
+        return (
+            Fore.RED
+            + f"Please set your {self.service_name} API key in .env or as an environment variable '{service_env}'.\n"
+            + f"You can get your key from {link} \n"
+            + "Alternatively, you can change the 'search_api' value in config.py to 'duckduckgo'."
+        )
 
 def check_config_setup() -> None:
```

Comment:
@foreztgump wouldn't it be better to check once depending on config selected search engine instead of calling all?